src/modules/reserve-overview/ReserveActions.tsx

```diff
@@ -74,35 +76,42 @@ export const ReserveActions = ({ reserve }: ReserveActionsProps) => {
     balance = walletBalances[API_ETH_MOCK_ADDRESS.toLowerCase()];
   }
 
-  const maxAmountToBorrow = getMaxAmountAvailableToBorrow(
-    reserve,
-    user,
-    InterestRate.Variable
-  ).toString();
+  let maxAmountToBorrow = getMaxAmountAvailableToBorrow(reserve, user, InterestRate.Variable);
+  const availableBorrowCap =
+    reserve.borrowCap === '0'
+      ? valueToBigNumber(ethers.constants.MaxUint256.toString())
+      : valueToBigNumber(reserve.borrowCap).minus(valueToBigNumber(reserve.totalDebt));
+  maxAmountToBorrow = BigNumber.max(BigNumber.min(maxAmountToBorrow, availableBorrowCap), 0);
 
   const maxAmountToBorrowUSD = amountToUSD(
-    maxAmountToBorrow,
+    maxAmountToBorrow.toString(),
     reserve.formattedPriceInMarketReferenceCurrency,
     marketReferencePriceInUsd
   );
 
-  const maxAmountToSupply = getMaxAmountAvailableToSupply(
+  let maxAmountToSupply = getMaxAmountAvailableToSupply(
     balance?.amount || '0',
     reserve,
     reserve.underlyingAsset,
     minRemainingBaseTokenBalance
-  ).toString();
+  );
+
+  const availableSupplyCap =
+    reserve.supplyCap === '0'
+      ? valueToBigNumber(ethers.constants.MaxUint256.toString())
+      : valueToBigNumber(reserve.supplyCap).minus(valueToBigNumber(reserve.totalLiquidity));
+  maxAmountToSupply = BigNumber.max(BigNumber.min(maxAmountToSupply, availableSupplyCap), 0);
```

Comment:
I was thinking with this change, the borrow modal would take into account the borrow cap when pressing the 'max' button to borrow, but that doesn't seem to be the case.

Reply:
Need to fix that too, good catch

Reply:
Should be factored into max amount in borrow modal

Reply:
oops I added this comment originally in the wrong spot...

But, should we also add the supply cap calculation to `getMaxAmountAvailableToSupply` and remove it from here? Then the supply modal would also get the correct max amount to supply, as well as the borrow modals.